src/azure-cli/azure/cli/command_modules/vm/operations/disk_encryption_set_identity.py

```diff
@@ -0,0 +1,185 @@
+from ..aaz.latest.disk_encryption_set import Update as _DiskEncryptionSetUpdate
+
+logger = get_logger(__name__)
+
+
+@register_command(
+    "disk-encryption-set identity assign",
+)
+class DiskEncryptionSetIdentityAssign(_DiskEncryptionSetUpdate):
+    """Add managed identities to an existing disk encryption set.
+
+    :example: Add a system assigned managed identity to an existing disk encryption set.
+        az disk-encryption-set identity assign --name MyDiskEncryptionSet --resource-group MyResourceGroup --system-assigned
+
+    :example: Add a user assigned managed identity to an existing disk encryption set.
+        az disk-encryption-set identity assign --name MyDiskEncryptionSet --resource-group MyResourceGroup --user-assigned MyAssignedId
+
+    :example: Add system assigned identity and a user assigned managed identity to an existing disk encryption set.
+        az disk-encryption-set identity assign --name MyDiskEncryptionSet --resource-group MyResourceGroup --system-assigned --user-assigned MyAssignedId
+    """
+
+    AZ_SUPPORT_GENERIC_UPDATE = False
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.system_assigned = AAZBoolArg(
+            options=["--system-assigned"],
+            help="Provide this flag to use system assigned identity.",
+            arg_group="Managed Identity",
+            blank=True
+        )
+        args_schema.user_assigned = AAZListArg(
+            options=["--user-assigned"],
+            help="User Assigned Identity ids to be used for disk encryption set. Accepts using the argument without any value.",
+            arg_group="Managed Identity",
+            blank=[]
+        )
+        args_schema.user_assigned.Element = AAZResourceIdArg(
+            fmt=AAZResourceIdArgFormat(template="/subscriptions/{subscription}/resourceGroups/{resource_group}"
+                                                "/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{}")
+        )
+        args_schema.no_wait._registered = False
+        args_schema.identity._registered = False
+        args_schema.enable_auto_key_rotation._registered = False
+        args_schema.federated_client_id._registered = False
+        args_schema.key_url._registered = False
+        args_schema.source_vault._registered = False
+        return args_schema
+
+    def pre_instance_update(self, instance):
+        existing_system_identity = False
+        existing_user_identity = set()
+        if instance.to_serialized_data().get('identity', None):
+            existing_system_identity = instance['identity']['type'].to_serialized_data() in ['SystemAssigned', 'SystemAssigned, UserAssigned']
+            existing_user_identity = {x.lower() for x in list(instance['identity'].to_serialized_data().get('userAssignedIdentities', {}).keys())}
```

Comment:
I'm not sure if the key in the instance if camel case or snake case
```suggestion
            existing_user_identity = {x.lower() for x in list(instance['identity'].to_serialized_data().get('user_assigned_identities', {}).keys())}
```